.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,181 @@
+                (
+                    startsWith(github.event.pull_request.title, 'perf:') ||
+                    startsWith(github.event.pull_request.title, 'Perf:') ||
+                    startsWith(github.event.pull_request.title, 'Performance:') ||
+                    startsWith(github.event.pull_request.title, 'performance:')
+                )
+            )
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                command: ["bench", "symfony", "rector"]
+                php: ["8.0", "8.1", "8.2", "8.3", "8.4", "8.5"]
+                xdebug_mode: ["no", "off", "develop", "coverage", "debug", "gcstats", "profile", "trace"]
+        steps:
+            -   uses: actions/checkout@v4
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    coverage: none
+                    tools: composer:v2.2.21
+
+            # ASLR can cause a lot of noise due to missed sse opportunities for memcpy
+            # and other operations, so we disable it during benchmarking.
+            -   name: Disable ASLR
+                run: echo 0 | sudo tee /proc/sys/kernel/randomize_va_space
+
+            -   name: Install valgrind
+                run: sudo apt-get update && sudo apt-get install -y valgrind
+                    
+            -   name: Compile
+                run: |
+                    ./.build.scripts/compile.sh
+                    sudo make install
+
+            -   name: create results and opcache folder
+                run: |
+                    mkdir -p results
+                    mkdir -p /tmp/opcache
+
+            -   name: install Symfony demo
+                if: matrix.command == 'symfony'   
+                run: |
+                    if [ "${{ matrix.php }}" \< "8.4" ]; then
+                        version="v2.0.2"
+                    else
+                        version="v2.7.0"
+                    fi
+                    git clone -q --branch "$version" --depth 1 https://github.com/symfony/demo.git ./symfony-demo
+                    cd symfony-demo
+                    composer install 
+                    find . -type f -name "*.php" -exec sed -i.bak -E 's/error_reporting\(.*\);/error_reporting(0);/g' {} +
+
+            -   name: install Rector
+                if: matrix.command == 'rector'   
+                run: |
+                    composer require rector/rector:2.1.2 --dev
+                    cp .github/benchmark-files/rector.php .
+
+            -   name: Copy ini file
+                run: |
+                    sudo cp ./.github/benchmark-files/benchmark.ini /etc/php/${{ matrix.php }}/cli/conf.d
+                    sudo cp ./.github/benchmark-files/benchmark.ini /etc/php/${{ matrix.php }}/cgi/conf.d
+
+            -   name: Copy xdebug ini file
+                if: matrix.xdebug_mode != 'no'   
```

Comment:
If Xdebug needs to be activated we also copy the Xdebug ini file to these folders